pip/pip-370.md

```diff
@@ -0,0 +1,91 @@
+  5. The config `allowAutoTopicCreation` still works for the local cluster as before, it will not be affected by the new config `createTopicToRemoteClusterForReplication`.
+- If `createTopicToRemoteClusterForReplication` is set to `true`.
+  a. All components work as before, see details: `Motivation -> The current topic creation behavior when enabling Geo-Replication`
+
+### Phase 2: The replicator will check remote topics' partitioned metadata and update partitions in the remote cluster to the same as the current cluster if needed.
+- If `createTopicToRemoteClusterForReplication` is set to `false`.
+  - The behavior is the same as Phase 1.
+- If `createTopicToRemoteClusterForReplication` is set to `true`.
+  - When a replicator for a topic partition starts, it checks the partitioned metadata in the remote cluster first and updates partitions in the remote cluster to the same as the current cluster if needed. Seem the example as follows:
+    -  `local_cluster.topic.partitions = 2` and the topic does not exist in the remote cluster: create a partitioned topic with `2` partitions in the remote cluster.
+      - Before `PIP-370 Phase 2`: the replicator will only trigger partition creation (`{topic}-partition-0` and `{topic}-partition-1`), and will not care about partitioned metadata.
+    -  `local_cluster.topic.partitions = 2` and `remote_cluster.topic.partitions = 1`: expand `remote_cluster.topic.partitions` to `2`.
+      - Before `PIP-370 Phase 2`: the replicator will only trigger partition creation (`{topic}-partition-0` and `{topic}-partition-1`), and the partitioned metadata in the remote cluster is still `1`. 
+    -  `local_cluster.topic.partitions = 2` and `remote_cluster.topic.partitions >=2 `: modifies nothing.
+      - Before `PIP-370 Phase 2`: It is the same as before `PIP-370`.
+    -  `local_cluster.topic.partitions = 2` and `remote_cluster.topic.partitions = 0(non-partitioned topic)`: the replicator prints error logs in the background.
+      -  Before `PIP-370 Phase 2`: the replicator will only trigger partition creation (`{topic}-partition-0` and `{topic}-partition-1`), then users will get `3` non-partitioned topics: `[{tp}, {topic}-partition-0, {topic}-partition-1`.
+    -  `local_cluster.topic.partitions = 0(non-partitioned topic)` and `remote_cluster.topic.partitions = 0(non-partitioned topic)`: modifies nothing.
+      - Before `PIP-370 Phase 2`: It is the same as before `PIP-370`.  
+    -  `local_cluster.topic.partitions = 0(non-partitioned topic)` and the topic does not exist in the remote cluster: create a non-partitioned topic in the remote cluster.
+      - Before `PIP-370 Phase 2`: It is the same as before `PIP-370`. 
+    -  `local_cluster.topic.partitions = 0(non-partitioned topic)` and and `remote_cluster.topic.partitions >= 1`: the replicator prints error logs in the background.
```

Comment:
Consider to add metrics so that users can get alerts for the case that can't start the data replicator.

Reply:
Added the sections: metrics and monitor